src/tox/venv.py

```diff
@@ -50,25 +50,32 @@ def readconfig(cls, path):
             for line in lines:
                 md5, depstring = line.split(None, 1)
                 deps.append((md5, depstring))
+            md5, python = value
             return CreationConfig(md5, python, version, sitepackages, usedevelop, deps, alwayscopy)
         except Exception:
             return None
 
-    def matches(self, other, deps_matches_subset=False):
-        return (
-            other
-            and self.md5 == other.md5
-            and self.python == other.python
-            and self.version == other.version
-            and self.sitepackages == other.sitepackages
-            and self.usedevelop == other.usedevelop
-            and self.alwayscopy == other.alwayscopy
-            and (
-                all(d in self.deps for d in other.deps)
-                if deps_matches_subset is True
-                else self.deps == other.deps
-            )
-        )
+    def matches(self, other, deps_matches_subset=False, provide_reason=False):
```

Comment:
this function (if annotated) would be `Union[Tuple[bool, str], bool]` -- does it make sense to instead have two functions: one without the reason and one with the reason?

Reply:
I would like to avoid duplication of the logic though.

Reply:
ah yeah you can just do:

```python
def whatever_with_reason(...):
    return True, 'reason'

def whatever_without_reason(...):
    ret, _ = whatever_with_reason(**kwargs)
    return ret
```

or something like that :man_shrugging:

Reply:
I see, my idea was that only makes the code more type correct; which is not really needed until we implement type annotations (probably once we drop Python 2 and 3.4).

Reply:
ah yeah I guess I didn't put my justifcation, my bad!

It would also be nice so we don't have a [boolean trap](https://ariya.io/2011/08/hall-of-api-shame-boolean-trap) parameter -- especially until there's enforced keyword-only-arguments in python3 they're a pretty easy pit to fall into in python where you accidentally pass too many positional arguments and it splats over a defaulted keyword argument